### ISSUE TITLE
gRPC & Kafka metadata

### ISSUE THREAD

**Issue by mmRoshani** [wenex-org/platform#2]
The Grpc header is the Map data structure although the Kafka header is kinda `Record<string, string>` I think it would be better to distinguish these so we get errors in the development phase and not in production. still, I change the functionality as I said before because I used the gRPC service call in my guards that I needed to be cached as well.

**Comment by vhidvz** [comment]
I appreciate your contribution Mojtaba, Map data structure and Record are very similar, especially in serialization. the exact type I used for metadata is `Record<string, any>` this is precisely like the `headers?: Record<string, any>` of Kafka messages.

**Comment by mmRoshani** [comment]
Yeah of course but there is a little problem since `any` may be a `Number` or even `Boolean` typed value and kafkaJS does not support that kind of value for headers(please take a look [here](https://kafka.js.org/docs/producing#message-headers)).
it would be nice if the type be `Record<string,string>` instead of `Record<string, any>`
